src/plugins/data/common/search/aggs/utils/get_aggs_formats.ts

```diff
@@ -133,11 +134,20 @@ export function getAggsFormats(getFieldFormat: GetFieldFormat): FieldFormatInsta
       static id = 'multi_terms';
       static hidden = true;
 
+      private formatCache: Map<SerializedFieldFormat<FieldFormatParams>, FieldFormat> = new Map();
+
       convert = (val: unknown, type: FieldFormatsContentType) => {
         const params = this._params;
-        const formats = (params.paramsPerField as SerializedFieldFormat[]).map((fieldParams) =>
-          getFieldFormat({ id: fieldParams.id, params: fieldParams })
-        );
+        const formats = (params.paramsPerField as SerializedFieldFormat[]).map((fieldParams) => {
+          const isCached = this.formatCache.has(fieldParams);
+          const cachedFormat =
+            this.formatCache.get(fieldParams) ||
+            getFieldFormat({ id: fieldParams.id, params: fieldParams });
+          if (!isCached) {
+            this.formatCache.set(fieldParams, cachedFormat);
+          }
```

Comment:
nit: inverting the order will remove few lines

```suggestion
          if (!this.formatCache.has(fieldParams);) {
            this.formatCache.set(
              fieldParams, 
              getFieldFormat({ id: fieldParams.id, params: fieldParams })
            );
          }
          return this.formatCache.get(fieldParams);
```